### PR TITLE
feat: add validation for cluster selector for override rule

### PR DESCRIPTION
**Pull request by britaniar**
### Description of your changes

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/utils/validator/clusterresourceoverride.go [hunk 1]
```diff
@@ -79,3 +90,20 @@ func validateClusterResourceOverrideResourceLimit(cro fleetv1alpha1.ClusterResou
...
+func validateClusterResourceOverrideRuleSelector(cro fleetv1alpha1.ClusterResourceOverride) error {
+	allErr := make([]error, 0)
+	for _, rule := range cro.Spec.Policy.OverrideRules {
+		if rule.ClusterSelector == nil {
+			continue
+		}
+		for _, selector := range rule.ClusterSelector.ClusterSelectorTerms {
+			// Check that only label selector is supported
+			if selector.LabelSelector == nil || selector.PropertySelector != nil || selector.PropertySorter != nil {
+				allErr = append(allErr, fmt.Errorf("label selector is only supported for resource selection %+v", selector))
```
Comment (zhiying-lin) [inline pkg/utils/validator/clusterresourceoverride.go:104]:
```suggestion
			if selector.PropertySelector != nil || selector.PropertySorter != nil {
				allErr = append(allErr, fmt.Errorf("invalid clusterSelector %v: only labelSelector is supported", selector))
```

then we check if the selector.LabelSelector is nil or not.

Comment (zhiying-lin) [reply pkg/utils/validator/clusterresourceoverride.go:104]: we need to call validateClusterSelector func to validate the selector?

File: pkg/utils/validator/clusterresourceoverride_test.go [hunk 2]
```diff
@@ -373,11 +403,41 @@ func TestValidateClusterResourceOverride(t *testing.T) {
 							Name:    "test-cluster-role",
 						},
 					},
+					Policy: validPolicy,
 				},
 			},
 			croList:    nil,
 			wantErrMsg: nil,
 		},
+		"invalid cluster resource override - fail validateClusterResourceOverrideRuleSelector": {
```
Comment (zhiying-lin) [inline pkg/utils/validator/clusterresourceoverride_test.go:412]: add more tests? eg, policy is nil, nil labelSelector, empty terms? empty matchLabels map and empty matchExpressions?

File: pkg/utils/validator/clusterresourceoverride.go [hunk 3]
```diff
@@ -79,3 +90,20 @@ func validateClusterResourceOverrideResourceLimit(cro fleetv1alpha1.ClusterResou
...
+			}
```
Comment (zhiying-lin) [inline pkg/utils/validator/clusterresourceoverride.go:105]: we also need to validate JSONPatchOverride to validate value https://github.com/Azure/fleet/blob/main/apis/placement/v1alpha1/override_types.go#L152

btw, the api will be changed in #748 and you can validate it in a separate PR once #748 is merged.

File: pkg/utils/validator/clusterresourceoverride.go [hunk 4]
```diff
@@ -79,3 +92,28 @@ func validateClusterResourceOverrideResourceLimit(cro fleetv1alpha1.ClusterResou
 	}
 	return errors.NewAggregate(allErr)
 }
+
+// validateClusterResourceOverridePolicy checks if override rule is selecting resource by name.
+func validateClusterResourceOverridePolicy(cro fleetv1alpha1.ClusterResourceOverride) error {
+	allErr := make([]error, 0)
+	for _, rule := range cro.Spec.Policy.OverrideRules {
```
Comment (ryanzhang-oss) [inline pkg/utils/validator/clusterresourceoverride.go:99]: isn't this the same as the resourceOverride? We should try to avoid code duplication

**Review by ryanzhang-oss — APPROVED** [review]

File: pkg/utils/validator/clusterresourceoverride_test.go [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
 
 import (
+	"errors"
 	"fmt"
 	"strings"
 	"testing"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/errors"
+	apiErrors "k8s.io/apimachinery/pkg/util/errors"
```
Comment (zhiying-lin) [inline pkg/utils/validator/clusterresourceoverride_test.go:10]: always use the lower case for the package name. please fix it in your next PR

File: pkg/utils/validator/clusterresourceoverride.go (hunk 3 above)
Comment (britaniar) [reply pkg/utils/validator/clusterresourceoverride.go]: Addressed in [#752 ](https://github.com/Azure/fleet/pull/752)
